snapshot-labs/sx-monorepo
# feat: configure and store multiple treasuries

### Summary

Closes: https://github.com/snapshot-labs/sx-monorepo/issues/18

Decisions:
1. We have two sources of truth for treasuries - executions and user configured treasuries on metadata, with this PR treasuries configured in metadata are used as a "whitelist" of executors that should be displayed on UI (in both treasury as well in proposal creation). You can also add treasuries that are not linked to executor to create read only treasury (can't be used in execution).
2. We had to index additional properties on backend to be able to link executors to treasuries.

### Screenshots

UI is temporary as design were not confirmed to be final.

<img width="626" alt="image" src="https://github.com/snapshot-labs/sx-monorepo/assets/1968722/3c197d12-f0e3-4658-99af-2caed00ab3d1">
<img width="603" alt="image" src="https://github.com/snapshot-labs/sx-monorepo/assets/1968722/fa24e8fb-7120-4c6f-81c7-1b470fa1848b">
<img width="770" alt="image" src="https://github.com/snapshot-labs/sx-monorepo/assets/1968722/f59e42b8-a772-4e42-9229-98c0c55fe837">


### How to test

1. Run `VITE_ENABLED_NETWORKS=sep yarn dev:full`. (it can take some time for subgraph to be in sync).
2. Interact with this space as a user (check treasury, try creating proposals): http://localhost:8080/#/sep:0x8b08aa6cdaf0bc92bb9cc6844459b2fed94249af/treasury
2. Create space on Sepolia with execution strategies (can't add them to old spaces as we don't have editor for it).
3. Set treasuries in metadata (so for timelock execution it's just strategy address, for avatar it's avatar's address).
